recipes/mender-client/debian-master/patches/boost-1.6.x-beast.patch

```diff
@@ -0,0 +1,132 @@
+Description: Patch the common/http implementation to go with Boost Beast 1.6.x
+Author: ole.orhagen@northern.tech
+Origin: other, <URL, required except if Author is present>
+Last-Update: 2023-05-11
+---
+This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
+Index: mender/common/http.hpp
+===================================================================
+--- mender.orig/common/http.hpp
++++ mender/common/http.hpp
+@@ -25,7 +25,7 @@
+ #ifdef MENDER_USE_BOOST_BEAST
+ #include <boost/asio.hpp>
+ #include <boost/beast.hpp>
+-#include <boost/beast/ssl.hpp>
++#include <boost/asio/ip/tcp.hpp>
+ #include <boost/asio/ssl/error.hpp>
+ #include <boost/asio/ssl/stream.hpp>
+ #endif // MENDER_USE_BOOST_BEAST
+@@ -48,6 +48,7 @@ namespace asio = boost::asio;
+ namespace beast = boost::beast;
+ namespace http = beast::http;
+ namespace ssl = asio::ssl;
++using tcp = boost::asio::ip::tcp; // from <boost/asio/ip/tcp.hpp>
+ #endif // MENDER_USE_BOOST_BEAST
+ 
+ namespace error = mender::common::error;
+@@ -314,7 +315,7 @@ private:
+ 
+ #ifdef MENDER_USE_BOOST_BEAST
+ 	boost::asio::ip::tcp::resolver resolver_;
+-	beast::ssl_stream<beast::tcp_stream> stream_;
++	ssl::stream<tcp::socket> stream_;
+ 
+ 	// This shared pointer is used as a workaround, points to ourselves, and has some peculiar
+ 	// properties. First the reason for the workaround: When calling `cancel()` on TCP streams,
+Index: mender/common/http/platform/beast/http.cpp
+===================================================================
+--- mender.orig/common/http/platform/beast/http.cpp
++++ mender/common/http/platform/beast/http.cpp
+@@ -14,7 +14,7 @@
+ 
+ #include <boost/asio/ip/tcp.hpp>
+ #include <boost/asio/ssl/verify_mode.hpp>
+-#include <boost/beast/core/stream_traits.hpp>
++#include <boost/beast/core/type_traits.hpp>
+ #include <boost/asio.hpp>
+ #include <common/http.hpp>
+ 
+@@ -94,7 +94,6 @@ Client::Client(ClientConfig &client, eve
+ 	resolver_(GetAsioIoContext(event_loop)),
+ 	stream_(GetAsioIoContext(event_loop), client.ctx_),
+ 	body_buffer_(HTTP_BEAST_BUFFER_SIZE) {
+-	response_buffer_.reserve(body_buffer_.size());
```

Comment:
Have you tested this? I suspect this will generate errors of the `need_buffer` kind.